homeassistant/components/light/tplink.py

```diff
@@ -7,40 +7,43 @@
 import logging
 import time
 
-import voluptuous as vol
-
-from homeassistant.const import (CONF_HOST, CONF_NAME)
 from homeassistant.components.light import (
     Light, ATTR_BRIGHTNESS, ATTR_COLOR_TEMP, ATTR_HS_COLOR, SUPPORT_BRIGHTNESS,
-    SUPPORT_COLOR_TEMP, SUPPORT_COLOR, PLATFORM_SCHEMA)
-import homeassistant.helpers.config_validation as cv
+    SUPPORT_COLOR_TEMP, SUPPORT_COLOR)
 from homeassistant.util.color import \
     color_temperature_mired_to_kelvin as mired_to_kelvin
 from homeassistant.util.color import (
     color_temperature_kelvin_to_mired as kelvin_to_mired)
+from homeassistant.components.tplink import DOMAIN as TPLINK_DOMAIN
+from homeassistant.exceptions import PlatformNotReady
+
+DEPENDENCIES = ['tplink']
 
-REQUIREMENTS = ['pyHS100==0.3.3']
+PARALLEL_UPDATES = 0
 
 _LOGGER = logging.getLogger(__name__)
 
 ATTR_CURRENT_POWER_W = 'current_power_w'
 ATTR_DAILY_ENERGY_KWH = 'daily_energy_kwh'
 ATTR_MONTHLY_ENERGY_KWH = 'monthly_energy_kwh'
 
-DEFAULT_NAME = 'TP-Link Light'
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_HOST): cv.string,
-    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
-})
 
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up discovered switches."""
+    from pyHS100 import SmartDeviceException
+    devs = []
+    for dev in hass.data[TPLINK_DOMAIN]['light']:
+        try:
+            # fetch MAC and name already now to avoid I/O inside init
```

Comment:
…

you're doing I/O inside an async method

Reply:
I am indeed. This used to be `setup_platform` so it used to be fine. So the correct way is to wrap this to be run in an executor, right?

Reply:
Changed to fetch the sysinfo during the initial `update()`, this simplifies the code as well.